pkg/reconciler/finalizer/base_finalizer.go

```diff
@@ -56,3 +67,48 @@ func (bf *baseFinalizer) destroy(ctx context.Context) status.MultiError {
 	klog.Info("Destroyer succeeded")
 	return nil
 }
+
+func (bf *baseFinalizer) unmanageObjects(ctx context.Context, rsyncRef client.ObjectKey) status.MultiError {
+	var errs status.MultiError
+	rg := &v1alpha1.ResourceGroup{}
+	if err := bf.Client.Get(ctx, rsyncRef, rg); err != nil {
+		if apierrors.IsNotFound(err) {
+			// ResourceGroup doesn't exist. This could be for a variety of reasons, but
+			// there's no work for the finalizer to do.
```

Comment:
Not entirely sure if this should succeed silently or not. 
Seems like we should at least log something.

I guess one case is that it hasn't successfully applied yet, so the RG hasn't been created, in which case it's fine to exit.

Are there other use cases where it might not?

Reply:
Does the reconciler-manager delete the RG after the RSync is NotFound or just after it's got a deletionTimestamp?

Reply:
> I guess one case is that it hasn't successfully applied yet, so the RG hasn't been created, in which case it's fine to exit. Are there other use cases where it might not?

The only happy path case I could think of is if the applier hasn't successfully run yet. It otherwise would be permanently wedged if some other client deleted the ResourceGroup, although nobody else _should_ delete the ResourceGroup

> Does the reconciler-manager delete the RG after the RSync is NotFound or just after it's got a deletionTimestamp?

It waits for the `reconciler` finalizer to complete first before executing the `reconciler-manager` finalizer.

Reply:
How about we check the RSync `status.lastSyncedCommit`. If it's set, then log an error. If it's not set, then log at info level. 

This might reduce visibility on some cases where the apply started, created the RG, and didn't succeed, but it should increase visibility on most other problematic cases where it's synced at least once before.